components/builder-http-gateway/src/http/middleware.rs

```diff
@@ -138,30 +138,49 @@ impl Authenticated {
-        let mut request = SessionGet::new();
-        request.set_token(token);
-        let conn = req.extensions.get_mut::<XRouteClient>().unwrap();
-        match conn.route::<SessionGet, Session>(&request) {
-            Ok(session) => {
-                self.validate_session(&session)?;
-                Ok(session)
+    fn authenticate(&self, req: &mut Request, token: &str) -> IronResult<Session> {
+        // Test hook - always create a valid session
+        if env::var_os("HAB_FUNC_TEST").is_some() {
+            debug!(
+                "HAB_FUNC_TEST: {:?}; calling session_create_short_circuit",
+                env::var_os("HAB_FUNC_TEST")
+            );
+            return session_create_short_circuit(req, token);
+        };
+
+        // Check for a valid personal access token
+        if bldr_core::access_token::is_access_token(token) {
+            match bldr_core::access_token::validate_access_token(&self.key_dir, token) {
+                Ok(session) => return Ok(session),
+                Err(bldr_core::Error::TokenExpired) => {
+                    let err = NetError::new(ErrCode::BAD_TOKEN, "net:auth:expired-token");
+                    return Err(IronError::new(err, Status::Forbidden));
+                }
+                Err(e) => {
+                    warn!("Unable to validate access token, err={:?}", e);
+                    let err = NetError::new(ErrCode::BAD_TOKEN, "net:auth:bad-token");
+                    return Err(IronError::new(err, Status::Forbidden));
+                }
             }
-            Err(err) => {
-                let status = net_err_to_http(err.get_code());
-                let body = itry!(serde_json::to_string(&err));
-                Err(IronError::new(err, (body, status)))
+        };
+
+        // Check for internal sessionsrv token
+        let decoded_token = match base64::decode(token) {
+            Ok(decoded_token) => decoded_token,
+            Err(e) => {
+                warn!("Failed to base64 decode token, err={:?}", e);
+                let err = NetError::new(ErrCode::BAD_TOKEN, "net:auth:decode");
+                return Err(IronError::new(err, Status::Forbidden));
             }
-        }
-    }
+        };
 
-    fn validate_session(&self, session: &Session) -> IronResult<()> {
-        let flags = FeatureFlags::from_bits(session.get_flags()).unwrap();
-        if !flags.contains(self.features) {
-            let err = NetError::new(ErrCode::ACCESS_DENIED, "net:auth:2");
-            return Err(IronError::new(err, Status::Forbidden));
+        if let Ok(session_token) = message::decode(&decoded_token) {
+            session_create(req, self.features, session_token)
+        } else {
+            // TBD: Deprecate github personal token support and remove this
+            // We are temporarily falling back to github personal access token
+            session_create_github(req, token)
```

Comment:
Is it worth outputting a log message here, maybe on `info` level, that says something to the effect of "GH auth token used - this is deprecated and will be removed"?  Basically, just something that is reminding us to remove this at a later date, so we don't forget about it.

Reply:
Good point - I'd rather not log here, as we'll get a huge amount of logs, but I'm thinking we might want to put a deprecation warning in the hab cli itself if a non-access token is being used.

Reply:
Sounds good.  I don't know how important it is for that kind of information to be communicated back via straight HTTP, e.g. if someone was hitting the API via curl.  It feels like a nice-to-have.  I tend to use curl a lot when testing the API but I'm not sure how much actual use it gets by our users.